FreeTubeApp/FreeTube
# Add support for lefthook local config for setting up option `rc`

Add support for lefthook local config for setting up option `rc`




## Pull Request Type


## Related issue
Improvement for lefthook added in #1788

## Description
I use SourceTree
And it's impossible to run lefthook in SourceTree due to ENV values in SourceTree are different from those in CLI
lefthook `1.2.4` introduced [`rc` option](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#rc)
Which allows GUI apps to also run lefthook with extra setup steps

This PR makes git ignores local config for lefthook `lefthook-local.yml` and an example file for it
Though due to [an issue (still present in `1.2.5`)](https://github.com/evilmartians/lefthook/pull/392#issuecomment-1350252074) the value must be absolute path, i.e. cannot use prefix `~/`

## Screenshots 
What do you expect to see ._.

## Testing 
- Install & Open SourceTree
- Switch to this branch
- Try committing a JS file, confirms it failed due to missing binary (might succeed for some setup but not for mine)
- Create `lefthook-local.yml` (just copy from `lefthook-local.yml.example`)
- Commit again, should work now

## Desktop

- **OS:** MacOS
- **OS Version:** 12.6.1
- **FreeTube version:** 29f5e38623d02f2e19a0d396aa615fbebf3d1238

## Additional context
